### ISSUE TITLE
Audio passthrough from app

### ISSUE THREAD

**Issue by testuser7** [jellyfin/jellyfin-tizen#278]
Hi, I have AV Receiver capable to decode most of the audio codecs connected over eARC HDMI with S90C TV.
When I start playing a video with TrueHD audio for short time I have information on AV side that TrueHD audio is played and after a second PCM signal is played.
Is it possible to passthrough audio from the app directly?

**Comment by dmitrylyzo** [comment]
You probably need to play with sound settings of TV. But, AFAIK, you can only pass through formats that your TV supports. So probably not.

One way to test (if the video codec is supported) - try playing the file from the USB drive connected to the TV.

**Comment by testuser7** [comment]
If I'm not mistaken, this TV model does not support TrueHD either.
There is passthrough option but only for HDMI inputs: https://us.community.samsung.com/t5/QLED-and-The-Frame-TVs/Sound-passthrough-option-greyed-out-while-using-tv-apps/td-p/2163302

**Comment by testuser7** [comment]
The TV appears to decode all supported audio formats and output them as:
1. for eARC:
 - AC-3 -> Multi Channel PCM
 - E-AC-3 -> Multi Channel PCM
 - E-AC-3 Atmos -> Multi Channel PCM with Atmos
 - E-AC-3 TrueHD Atmos - > Multi Channel PCM with Atmos
2. for ARC:
 - AC-3 -> E-AC-3
 - E-AC-3 -> E-AC-3
 - E-AC-3 Atmos -> E-AC-3
 - E-AC-3 TrueHD Atmos - > E-AC-3

I haven't tested the optical output.
Dolby TrueHD (lossless) is not supported.

**Comment by testuser7** [comment]
In the latest version of `jellyfin-web` there is an option to set the priority for audio codecs: https://github.com/jellyfin/jellyfin-web/pull/5434
After setting the `Opus` codec the audio is still transcoded to `AAC` even though the TV supports this codec: https://developer.samsung.com/smarttv/develop/specifications/media-specifications/2023-tv-video-specifications.html

Are any changes needed on the `jellyfin-tizen` side to be able to use the `Opus` codec?

**Comment by dmitrylyzo** [comment]
> Are any changes needed on the `jellyfin-tizen` side to be able to use the `Opus` codec?

No. It is checked [here](https://github.com/jellyfin/jellyfin-web/blob/8ceb8e14dd8ac086f959f0208b1f1d1e9b727dcb/src/scripts/browserDeviceProfile.js#L163-L169). For Tizen, `canPlayType` is used ([1](https://github.com/jellyfin/jellyfin-web/blob/8ceb8e14dd8ac086f959f0208b1f1d1e9b727dcb/src/scripts/browserDeviceProfile.js#L169) + [2](https://github.com/jellyfin/jellyfin-web/blob/8ceb8e14dd8ac086f959f0208b1f1d1e9b727dcb/src/scripts/browserDeviceProfile.js#L187)), but it may lie.
Iirc, there is [report](https://github.com/jellyfin/jellyfin-tizen/issues/130) that OPUS doesn't work on some models, presumably in TS (https://github.com/jellyfin/jellyfin-tizen/issues/130#issuecomment-1208110002).

**Comment by testuser7** [comment]
> > Are any changes needed on the `jellyfin-tizen` side to be able to use the `Opus` codec?
> 
> No. It is checked [here](https://github.com/jellyfin/jellyfin-web/blob/8ceb8e14dd8ac086f959f0208b1f1d1e9b727dcb/src/scripts/browserDeviceProfile.js#L163-L169). For Tizen, `canPlayType` is used ([1](https://github.com/jellyfin/jellyfin-web/blob/8ceb8e14dd8ac086f959f0208b1f1d1e9b727dcb/src/scripts/browserDeviceProfile.js#L169) + [2](https://github.com/jellyfin/jellyfin-web/blob/8ceb8e14dd8ac086f959f0208b1f1d1e9b727dcb/src/scripts/browserDeviceProfile.js#L187)), but it may lie. Iirc, there is [report](https://github.com/jellyfin/jellyfin-tizen/issues/130) that OPUS doesn't work on some models, presumably in TS ([#130 (comment)](https://github.com/jellyfin/jellyfin-tizen/issues/130#issuecomment-1208110002)).

Thanks for the info.
I wonder if it would be possible to decode the audio and stream as LPCM.
Can ffmpeg stream Atmos data with LPCM?

**Comment by testuser7** [comment]
@dmitrylyzo I just checked and on the latest `2013` firmware (Tizen 8.0) I can play Dolby TrueHD (loseless) on my TV from an external source via HDMI (Kodi).
But the same format still doesn't work in Jellyfin.

**Comment by dmitrylyzo** [comment]
> [@dmitrylyzo](https://github.com/dmitrylyzo) I just checked and on the latest `2013` firmware (Tizen 8.0) I can play Dolby TrueHD (loseless) on my TV from an external source via HDMI (Kodi). But the same format still doesn't work in Jellyfin.

Are you sure Kodi is transmitting TrueHD to the TV and not transcoding it to AC-3/E-AC-3?

**Comment by testuser7** [comment]
> Are you sure Kodi is transmitting TrueHD to the TV and not transcoding it to AC-3/E-AC-3?

Yes, I checked the info menu and it shows `DTHD`.

**Comment by testuser7** [comment]
> I still think the only proper way to test is to play video from the USB on the native player.

I checked USB playback and:
`The audio data is not supported.`

It's not clear to me why it works via HDMI...

> Have you tried it?

Yes, I had that option checked.
